library/src/test/java/org/apache/apex/malhar/lib/state/spillable/SpillableByteMapImplTest.java

```diff
@@ -411,45 +441,48 @@ public void recoveryTest()
     testMeta.store.setup(testMeta.operatorContext);
     map1.setup(testMeta.operatorContext);
 
-    map1.beginWindow(1000);
+    System.out.println("0");
+    testMeta.store.beginWindow(0);
+    map1.beginWindow(0);
     map1.put("x", "1");
     map1.put("y", "2");
     map1.put("z", "3");
     map1.endWindow();
-    map1.beginWindow(1001);
+    testMeta.store.endWindow();
+
+    System.out.println("1");
+    testMeta.store.beginWindow(1);
+    map1.beginWindow(1);
     map1.put("x", "4");
     map1.put("y", "5");
     map1.endWindow();
-
-    testMeta.store.beforeCheckpoint(1001);
-    testMeta.store.checkpointed(1001);
+    testMeta.store.endWindow();
+    testMeta.store.beforeCheckpoint(1);
+    testMeta.store.checkpointed(1);
 
     SpillableByteMapImpl<String, String> clonedMap1 = KryoCloneUtils.cloneObject(map1);
 
-    map1.beginWindow(1002);
-    map1.put("x", "6");
-    map1.put("y", "7");
-    map1.endWindow();
-
-    Assert.assertEquals("6", map1.get("x"));
-    Assert.assertEquals("7", map1.get("y"));
-    Assert.assertEquals("3", map1.get("z"));
-
-    map1.beginWindow(1003);
-    map1.put("x", "8");
-    map1.put("y", "9");
+    System.out.println("2");
+    testMeta.store.beginWindow(2);
+    map1.beginWindow(2);
+    map1.put("x1", "6");
+    map1.put("y1", "7");
     map1.endWindow();
+    testMeta.store.endWindow();
 
     // simulating crash here
     map1.teardown();
     testMeta.store.teardown();
 
+    System.out.println("Recovering");
+
     map1 = clonedMap1;
     map1.getStore().setup(testMeta.operatorContext);
     map1.setup(testMeta.operatorContext);
```

Comment:
If activation window = -1, then ManagedState will not care about the data saved after the activation window